tests/conftest.py

```diff
@@ -1,11 +1,100 @@
+#   all (try finding every supported array library available in the environment)
+GLASS_ARRAY_BACKEND: str | bool = os.environ.get("GLASS_ARRAY_BACKEND", False)
+
+
+def _check_version(lib: str, array_api_compliant_version: str) -> None:
+    """
+    Check if installed library's version is compliant with the array API standard.
+
+    Parameters
+    ----------
+    lib
+        name of the library.
+    array_api_compliant_version
+        version of the library compliant with the array API standard.
+
+    Raises
+    ------
+    ImportError
+        If the installed version is not compliant with the array API standard.
+    """
+    lib_version = packaging.version.Version(importlib.metadata.version(lib))
+    if lib_version < packaging.version.Version(array_api_compliant_version):
+        msg = f"{lib} must be >= {array_api_compliant_version}; found {lib_version}"
+        raise ImportError(msg)
+
+
+def _import_and_add_numpy(xp_available_backends: dict[str, types.ModuleType]) -> None:
+    """Add numpy to the backends dictionary."""
+    _check_version("numpy", "2.1.0")
+    xp_available_backends.update({"numpy": np})
+
+
+def _import_and_add_array_api_strict(
+    xp_available_backends: dict[str, types.ModuleType],
+) -> None:
+    """Add array_api_strict to the backends dictionary."""
+    import array_api_strict
+
+    _check_version("array_api_strict", "2.0.0")
+    xp_available_backends.update({"array_api_strict": array_api_strict})
+    array_api_strict.set_array_api_strict_flags(api_version="2023.12")
+
+
+def _import_and_add_jax(xp_available_backends: dict[str, types.ModuleType]) -> None:
+    """Add jax to the backends dictionary."""
+    import jax
+
+    _check_version("jax", "0.4.32")
+    xp_available_backends.update({"jax.numpy": jax.numpy})
+    # enable 64 bit numbers
+    jax.config.update("jax_enable_x64", val=True)
+
+
+# a dictionary with all array backends to test
+xp_available_backends: dict[str, types.ModuleType] = {}
+
+# if no backend passed, use numpy by default
+if not GLASS_ARRAY_BACKEND or GLASS_ARRAY_BACKEND == "numpy":
+    _import_and_add_numpy(xp_available_backends)
+elif GLASS_ARRAY_BACKEND == "array_api_strict":
+    _import_and_add_array_api_strict(xp_available_backends)
+elif GLASS_ARRAY_BACKEND == "jax":
+    _import_and_add_jax(xp_available_backends)
+# if all, try importing every backend
+elif GLASS_ARRAY_BACKEND == "all":
+    with contextlib.suppress(ImportError):
+        _import_and_add_numpy(xp_available_backends)
+
+    with contextlib.suppress(ImportError):
+        _import_and_add_array_api_strict(xp_available_backends)
+
+    with contextlib.suppress(ImportError):
+        _import_and_add_jax(xp_available_backends)
+else:
+    msg = f"unsupported array backend: {GLASS_ARRAY_BACKEND}"
+    raise ValueError(msg)
+
+# use this as a decorator for tests involving array API compatible functions
+array_api_compatible = pytest.mark.parametrize("xp", xp_available_backends.values())
```

Comment:
In my opinion all the common testing utilities should go directly into `conftest.py` because that is the sole use of this file (even if it gets crowded).

Reply:
Fair enough. I'm just concerned that it can make it harder for others to contribute. Would be good for it to be as neat and granular as possible.